src/tools/kotlin-refinement-generator.ts

```diff
@@ -84,9 +82,6 @@ class KotlinRefinementGenerator extends RefinementExpressionVisitor<string> {
     }
     return `${expr.value.toString()}.asExpr()`;
   }
-  visitBooleanPrimitive(expr: BooleanPrimitive): string {
```

Comment:
This seemed wrong in the first place, since it should have been ${expr.value}.asExpr(), but where is boolean handled now?